paper-api/src/main/java/io/papermc/paper/command/brigadier/argument/resolvers/ObjectiveResolver.java

```diff
@@ -0,0 +1,66 @@
+package io.papermc.paper.command.brigadier.argument.resolvers;
+
+import com.mojang.brigadier.exceptions.CommandSyntaxException;
+import io.papermc.paper.command.brigadier.CommandSourceStack;
+import io.papermc.paper.command.brigadier.argument.ArgumentTypes;
+import org.bukkit.Bukkit;
+import org.bukkit.scoreboard.Objective;
+import org.bukkit.scoreboard.Scoreboard;
+import org.jetbrains.annotations.ApiStatus;
+import org.jspecify.annotations.NullMarked;
+
+/**
+ * A resolver that's capable of resolving
+ * an {@link Objective} value using a {@link Scoreboard} and {@link CommandSourceStack}.
+ *
+ * @see ArgumentTypes#objective()
+ */
+@ApiStatus.Experimental
+@ApiStatus.NonExtendable
+@NullMarked
+public interface ObjectiveResolver {
```

Comment:
Probably better to have boolean overloads here?
The only thing resolveWritable does is one more check and a nice exception.
Having it named differently feels confusing.